evalml/data_checks/datetime_format_data_check.py

```diff
@@ -378,110 +348,147 @@ def validate(self, X, y):
             )
             return messages
 
-        # Check if the data is monotonically increasing
-        no_nan_datetime_values = datetime_values.dropna()
-        if not pd.DatetimeIndex(no_nan_datetime_values).is_monotonic_increasing:
-            messages.append(
-                DataCheckError(
-                    message="Datetime values must be sorted in ascending order.",
-                    data_check_name=self.name,
-                    message_code=DataCheckMessageCode.DATETIME_IS_NOT_MONOTONIC,
-                ).to_dict(),
-            )
-
-        col_name = (
-            self.datetime_column if self.datetime_column != "index" else "either index"
-        )
-
-        ww_payload = infer_frequency(
-            pd.Series(datetime_values),
-            debug=True,
-            window_length=4,
-            threshold=0.4,
+        series_datetime = (
+            [datetime_values] if self.series_id is None else X[self.series_id].unique()
         )
-        inferred_freq = ww_payload[0]
-        debug_object = ww_payload[1]
-        if inferred_freq is not None:
-            return messages
-
-        # Check for NaN values
-        if len(debug_object["nan_values"]) > 0:
-            messages.append(
-                DataCheckError(
-                    message=f"Input datetime column '{col_name}' contains NaN values. Please impute NaN values or drop these rows.",
-                    data_check_name=self.name,
-                    message_code=DataCheckMessageCode.DATETIME_HAS_NAN,
-                ).to_dict(),
+        for series in series_datetime:
+            # if multiseries only select the datetimes corresponding to one series
+            if is_multiseries:
+                if self.datetime_column != "index":
+                    datetime_values = X[X[self.series_id] == series][
+                        self.datetime_column
+                    ].reset_index(drop=True)
+                else:
+                    datetime_values = X[X[self.series_id] == series].index
+
+            # Check if the data is monotonically increasing
+            no_nan_datetime_values = datetime_values.dropna()
+            if not pd.DatetimeIndex(no_nan_datetime_values).is_monotonic_increasing:
+                messages.append(
+                    DataCheckError(
+                        message="Datetime values must be sorted in ascending order.",
```

Comment:
Unsure if I need to change the message here to say something like "Datetime values {at series x} must be sorted in ascending order" or if it's alright to just keep it general.